src/core/unit.h

```diff
@@ -460,6 +460,9 @@ typedef struct Unit {
         bool exported_log_ratelimit_interval:1;
         bool exported_log_ratelimit_burst:1;
 
+        /* Whether we've seen a pidfd during deserialization of this unit. */
+        bool deserialize_got_pidfd:1;
```

Comment:
why is this a per-unit field? why not a global field?

Reply:
Sure I can make it a global field.